acceptance/tests/resource/file/ticket_6448_file_with_utf8_source.rb

```diff
@@ -0,0 +1,82 @@
+require 'puppet/acceptance/environment_utils'
+extend Puppet::Acceptance::EnvironmentUtils
+
+test_name 'Ensure a file resource can have a UTF-8 source attribute, content, and path when served via a module' do
+  skip_test 'requires a master for serving module content' if master.nil?
+
+  tmp_environment = mk_tmp_environment_with_teardown(master, File.basename(__FILE__, '.*'))
+  agent_tmp_dirs = {}
+
+  agents.each do |agent|
+    agent_tmp_dirs[agent.hostname] = agent.tmpdir(tmp_environment)
+  end
+
+  teardown do
+    step 'remove all test files on agents' do
+      agents.each { |agent| on(agent, "rm #{agent_tmp_dirs[agent.hostname]}", :accept_all_exit_codes => true) }
+    end
+    # note - master teardown is registered by #mk_tmp_environment_with_teardown
+  end
+
+  step 'create unicode source file served via module on master' do
+    # 静 \u9759 0xE9 0x9D 0x99 http://www.fileformat.info/info/unicode/char/9759/index.htm
+    # 的 \u7684 0xE7 0x9A 0x84 http://www.fileformat.info/info/unicode/char/7684/index.htm
+    # ☃ \2603 0xE2 0x98 0x83 http://www.fileformat.info/info/unicode/char/2603/index.htm
+    setup_module_on_master = <<-MASTER_MANIFEST
+      File {
+        ensure => directory,
+        mode => "0755",
+      }
+
+      file {
+        '#{environmentpath}/#{tmp_environment}/modules/utf8_file_module':;
+        '#{environmentpath}/#{tmp_environment}/modules/utf8_file_module/files':;
+      }
+      
+      file { '#{environmentpath}/#{tmp_environment}/modules/utf8_file_module/files/\u9759\u7684':
+        ensure => file,
+        content => "\u2603"
+      }
+    MASTER_MANIFEST
+    apply_manifest_on(master, setup_module_on_master, :expect_changes => true)
```

Comment:
Good question. I looked in other acceptance tests and modeled this approach after the direct puppet tests, ex here: https://github.com/puppetlabs/puppet/blob/master/acceptance/tests/direct_puppet/static_catalog_env_control.rb#L19-L154, which lay down the files in this manner rather than using fixtures. There's a tradeoff - I think there would be less complexity in this test if we could just copy the files over from fixtures, for sure. But I also think there's a downside, in that a "later-us" trying to understand the test potentially has to go searching around many places in the code to understand what it's doing, and when we set up shared contexts we have less flexibility to change them later as they may be used by other tests. There's probably an in-between that would suffice for this test, but I'm actually inclined to leave as-is, if not just so that others looking to write utf-8 file resource acceptance tests can see this as example content all in one place.